### DISCUSSION TITLE
Only gpt-3.5-turbo works. Older models return 404 error.

### DISCUSSION THREAD

**Discussion by jacksutherland** [tectalichq/public-openai-client-php#25] [Q&A] [answered]
I've change the model to multiple options gpt-3.5-turbo, text-davinci-003, text-davinci-002, etc, and gpt-3.5-turbo is the only one that returns a valid response. I get a 404 for everything else. I'm happy to accept it's something on my part, except I've gone trough the docs verbatim.

This wouldn't be a issue except that gpt-3.5-turbo is _extremely_ slow. Sometimes a 5-10 second response, which will result in a bounce for most users. So I'd like to use an older/faster model where possible.

The Stack Trace:

```
**Tectalic\OpenAi\ClientException**
Unsuccessful response. HTTP status code: 404 (Not Found).

1. in .../tectalic/openai/src/Handlers/ChatCompletions.phpat line 150
141142143144145146147148149150151152153154155156157158159     *
     * @api
     * @return AbstractModel|AbstractModelCollection
     * @throws ClientException if an unsuccessful HTTP response occurs.
     * @throws ClientException if the response body cannot be parsed.
     */
    public function toModel(): object
    {
        if ($this->getResponse()->getStatusCode() < 200 || $this->getResponse()->getStatusCode() >= 300) {
            throw new ClientException(
                \sprintf(
                    'Unsuccessful response. HTTP status code: %s (%s).',
                    $this->getResponse()->getStatusCode(),
                    $this->getResponse()->getReasonPhrase()
                )
            );
        }
        $class = $this->modelType;
        try {
```

**Comment by thejamescollins** [comment, marked as answer]
Hi there @jacksutherland,

Some of those models are only available via the `chatCompletions` handler, and others are available via the `completions` handler.

Please see the [Examples in the readme](https://github.com/tectalichq/public-openai-client-php#examples) for details on which models are available on which endpoints.

You may also find [this official information from OpenAI](https://platform.openai.com/docs/models/model-endpoint-compatibility) useful too.

Regarding API endpoint speeds (latencies), you may like to [review these suggestions from OpenIA themselves](https://platform.openai.com/docs/guides/production-best-practices/improving-latencies).

All the best,

James

**Reply by jacksutherland** [reply]
Ah, I see. Your [readme](https://github.com/tectalichq/public-openai-client-php#examples) lists `text-davinci-003` and `text-davinci-002` as supported for the `completions` and `chatCompletions` handlers, but the Open AI doc only shows them available for the `completions` endpoint.

**Reply by thejamescollins** [reply]
Thanks @jacksutherland, we have now corrected this error in our readme (https://github.com/tectalichq/public-openai-client-php/pull/27).
